### PR TITLE
room with screens: added, fixed

### PR DESCRIPTION
**Pull request by Razor0744**
favourite screen: added

### CODE REVIEW

File: app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+    override suspend fun getSchedules(): List<List<AnimeSeries>> = withContext(Dispatchers.IO) {
+        val localSchedules = animeSchedulesDao.getAnimeSchedules()
+
+        when {
+            listSchedules.isNotEmpty() -> {
+                listSchedules
+            }
+
+            localSchedules.isNotEmpty() -> {
+                for (i in 0..6) {
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt]: magic number anti-pattern.

File: app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt [hunk 2]
```diff
@@ -1,15 +1,61 @@
...
+    private val animeSchedulesDao: AnimeSchedulesDao
+) : AnilibriaRepository {
 
-    override suspend fun getSchedules(): List<AnilibriaSchedule> {
-        val response = anilibriaService.getSchedule()
+    private var listSchedules: List<List<AnimeSeries>> = emptyList()
 
-        return if (response.isSuccessful) response.body() ?: emptyList() else emptyList()
+    override suspend fun getSchedules(): List<List<AnimeSeries>> = withContext(Dispatchers.IO) {
+        val localSchedules = animeSchedulesDao.getAnimeSchedules()
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt]: Implementation of in-memory cache is incorrect: you always going to the db to check data. Also, please add mutex

File: app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt [hunk 3]
```diff
@@ -1,16 +1,92 @@
...
-    AnilibriaRepository {
+class AnilibriaRepositoryImpl(
+    private val anilibriaService: AnilibriaService,
+    private val detailsStateDao: DetailsStateDao
+) : AnilibriaRepository {
 
-    override suspend fun getTitle(id: Int): AnilibriaTitle? = withContext(Dispatchers.IO) {
-        val response = anilibriaService.getTitle(id = id)
-        if (response.isSuccessful) response.body() as AnilibriaTitle else null
+    private var state: DetailsState? = null
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt]: Something strange with model name and variable name. Why do you have state on data layer?

File: app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt [hunk 4]
```diff
@@ -1,16 +1,92 @@
...
+    private var idState: Int? = null
+
+    override suspend fun getTitle(id: Int): DetailsState? = withContext(Dispatchers.IO) {
+        val localState = detailsStateDao.getTitleById(id = id, category = API_CATEGORY_ANILIBRIA)
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt]: the same according to in-memory cache

File: app/src/main/java/com/example/movopfy/features/details/data/repository/KinopoiskRepositoryImpl.kt [hunk 5]
```diff
@@ -1,14 +1,71 @@
...
 
-class KinopoiskRepositoryImpl(private val kinopoiskService: KinopoiskService): KinopoiskRepository {
+class KinopoiskRepositoryImpl(
+    private val kinopoiskService: KinopoiskService,
+    private val detailsStateDao: DetailsStateDao
+) : KinopoiskRepository {
 
-    override suspend fun getTitle(id: Int): KinopoiskTitle? {
-        val response =  kinopoiskService.getTitle(id = id)
+    private var state: DetailsState? = null
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/details/data/repository/KinopoiskRepositoryImpl.kt]: The same here

File: app/src/main/java/com/example/movopfy/features/details/domain/models/DetailsState.kt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.example.movopfy.features.details.domain.models
+
+import com.example.movopfy.network.anilibria.models.AnilibriaEpisodesList
+
+data class DetailsState(
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/details/domain/models/DetailsState.kt:5]: The word "state" here is confusing

File: app/src/main/java/com/example/movopfy/features/details/presentation/viewmodel/DetailsViewModel.kt [hunk 7]
```diff
@@ -2,61 +2,74 @@ package com.example.movopfy.features.details.presentation.viewmodel
...
     sealed interface DetailsUiState {
 
         data object Loading : DetailsUiState
 
         data class Loaded(
-            val anilibriaTitle: AnilibriaTitle?,
-            val kinopoiskTitle: KinopoiskTitle?,
-            val episodesList: List<AnilibriaEpisodesList>?
+            val detailsState: DetailsState?,
+            val roomFavourite: RoomFavourite?
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/details/presentation/viewmodel/DetailsViewModel.kt]: Room model from database model? Please create mapper for it and dont use model from data layer in presentation layer

File: app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteItem.kt [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import com.example.movopfy.features.favourite.presentation.viewmodel.FavouriteViewModel
+import com.example.movopfy.uiComponents.theme.dimensions
+
+@Composable
+fun FavouriteItem(
+    modifier: Modifier = Modifier,
+    item: RoomFavourite,
+    viewModel: FavouriteViewModel
+) {
+    println(item)
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteItem.kt]: please remove

File: app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteScreen.kt [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import com.example.movopfy.features.favourite.presentation.viewmodel.FavouriteViewModel
+import com.example.movopfy.uiComponents.components.ProgressBarLoading
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+fun FavouriteScreen(
+    modifier: Modifier = Modifier,
+    viewModel: FavouriteViewModel = koinViewModel()
+) {
+
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteScreen.kt]: please remove empty line

File: app/src/main/java/com/example/movopfy/features/home/data/repository/KinopoiskRepositoryImpl.kt [hunk 10]
```diff
@@ -1,23 +1,59 @@
...
+) : KinopoiskRepository {
+
+    private var kinopoiskDocsList: List<KinopoiskDocs> = emptyList()
 
     override suspend fun getList(page: Int, category: String): List<KinopoiskDocs> =
         withContext(Dispatchers.IO) {
-            val response = kinopoiskService.getList(page = page, category = category)
-            if (response.isSuccessful) response.body()?.docs ?: emptyList() else emptyList()
-        }
+            val localList = kinopoiskDocsDao.getKinopoiskDocsByCategory(category = category)
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/home/data/repository/KinopoiskRepositoryImpl.kt]: the same problem

File: app/src/main/java/com/example/movopfy/uiComponents/components/FavouriteIcon.kt [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    modifier: Modifier = Modifier,
+    isFavorite: Boolean,
+    onIconClick: () -> Unit
+) {
+    if (!isFavorite) {
+        Icon(
+            modifier = modifier
+                .clickable { onIconClick() },
+            painter = painterResource(id = R.drawable.ic_star),
+            contentDescription = "is_favorite_false",
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/uiComponents/components/FavouriteIcon.kt]: dont add content description like this

Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/uiComponents/components/FavouriteIcon.kt]: did

File: app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt (hunk 1 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt]: did

File: app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteItem.kt (hunk 8 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteItem.kt]: did

File: app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteScreen.kt (hunk 9 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/favourite/presentation/ui/FavouriteScreen.kt]: did

File: app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt (hunk 2 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/anime/data/repository/AnilibriaRepositoryImpl.kt]: did

File: app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt (hunk 4 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt]: did

File: app/src/main/java/com/example/movopfy/features/home/data/repository/KinopoiskRepositoryImpl.kt (hunk 10 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/home/data/repository/KinopoiskRepositoryImpl.kt]: did

File: app/src/main/java/com/example/movopfy/features/details/data/repository/KinopoiskRepositoryImpl.kt (hunk 5 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/details/data/repository/KinopoiskRepositoryImpl.kt]: did

File: app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt (hunk 3 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/details/data/repository/AnilibriaRepositoryImpl.kt]: did

File: app/src/main/java/com/example/movopfy/features/details/domain/models/DetailsState.kt (hunk 6 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/details/domain/models/DetailsState.kt:5]: fixed

File: app/src/main/java/com/example/movopfy/features/details/presentation/viewmodel/DetailsViewModel.kt (hunk 7 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/details/presentation/viewmodel/DetailsViewModel.kt]: did

**Review by MaximSemashko — APPROVED** [review]

**Pull request merged**